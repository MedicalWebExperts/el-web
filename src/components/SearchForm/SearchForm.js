import React, { Component } from 'react';
import {
  arrayOf, shape, func, string,
} from 'prop-types';

import Input from '../Input/Input';
import Select from '../Select/Select';
import Button from '../Button/Button';
import media from '../../utils/media';

class SearchForm extends Component {
  static propTypes = {
    fields: arrayOf(shape({})).isRequired,
    onSearch: func.isRequired,
    styles: shape({}),
    buttonText: string,
  };

  static defaultProps = {
    styles: {},
    buttonText: 'Search',
  };

  state = {
    searchValues: [],
    advancedSearch: false,
  };

  handleChange = (selectedItem) => {
    const { name, value } = selectedItem;
    this.setState(prevState => ({
      searchValues: Object.assign({}, prevState.searchValues, { [name]: value }),
    }));
  };

  handleSearch = () => {
    this.props.onSearch(this.state.searchValues);
  };

  showAdvancedSearch = () => {
    this.setState(prevState => ({ advancedSearch: !prevState.advancedSearch }));
  };

  render() {
    const { fields, buttonText, styles } = this.props;
    const { advancedSearch } = this.state;

    return (
      <div style={styles}>
        <style jsx>
          {`
            div {
              display: flex;
              flex-direction: ${media.isMobile() || media.isTablet() ? 'column' : 'row'};
              justify-content: space-evenly;
              width: 100%;
            }
          `}
        </style>
        {fields.map((field, key) => {
          if (field.type === 'input') {
            return (
              <Input
                key={key}
                style={field.styles}
                type="text"
                name={field.name}
                value={field.value}
                onChange={this.handleChange}
                placeholder={field.placeholder}
              />
            );
          }
          return (
            (((media.isMobile() || media.isTablet()) && advancedSearch)
              || ((media.isTabletLanscape() || media.isFullScreen()) && !advancedSearch)) && (
              <Select
                key={key}
                options={field.options}
                onChange={this.handleChange}
                styles={field.styles}
                placeholder={field.placeholder}
              />
            )
          );
        })}
        {(media.isMobile() || media.isTablet()) && (
          <Button outline onClick={this.showAdvancedSearch} text="Advanced Search" />
        )}
        <Button onClick={this.handleSearch} text={buttonText} />
      </div>
    );
  }
}

export default SearchForm;
