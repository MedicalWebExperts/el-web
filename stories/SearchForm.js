import React from 'react';
import { storiesOf } from '@storybook/react';

import { SearchForm } from '..';

const fields = [
  {
    type: 'input',
    styles: {},
    name: 'name',
    placeholder: 'name',
  },
  {
    type: 'select',
    placeholder: 'animals',
    options: [
      { name: 'Tiger', value: 'tiger' },
      { name: 'Lion', value: 'lion' },
      { name: 'Panther', value: 'panther' },
      { name: 'Jaguar', value: 'jaguar' },
    ],
  },
  {
    type: 'select',
    placeholder: 'vehicles',
    options: [
      { name: 'Car', value: 'car' },
      { name: 'Plane', value: 'plane' },
      { name: 'Bus', value: 'bus' },
      { name: 'Bycicle', value: 'bycicle' },
    ],
  },
  {
    type: 'select',
    placeholder: 'cities',
    options: [
      { name: 'London', value: 'london' },
      { name: 'Cardiff', value: 'cardiff' },
      { name: 'Iowa', value: 'iowa' },
      { name: 'Bogota', value: 'bogota' },
    ],
  },
  {
    type: 'select',
    placeholder: 'gender',
    options: [{ name: 'Male', value: 'male' }, { name: 'Female', value: 'female' }],
  },
];

export default () => {
  storiesOf('SearchForm', module).add('Default', () => (
    <SearchForm
      onSearch={values => console.log('Search Form values: ', values)}
      fields={fields}
      buttonText="Find Doctor"
    />
  ));
};
