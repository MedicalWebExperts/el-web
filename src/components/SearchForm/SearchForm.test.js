import React from 'react';
import renderer from 'react-test-renderer';

import SearchForm from './SearchForm';

const fields = [
  {
    type: 'input',
    styles: {},
    name: 'name',
    placeholder: 'name',
  },
  {
    type: 'select',
    placeholder: 'animals',
    options: [
      { name: 'Tiger', value: 'tiger' },
      { name: 'Lion', value: 'lion' },
      { name: 'Panther', value: 'panther' },
      { name: 'Jaguar', value: 'jaguar' },
    ],
  },
  {
    type: 'select',
    placeholder: 'gender',
    options: [{ name: 'Male', value: 'male' }, { name: 'Female', value: 'female' }],
  },
];

const props = {
  fields,
  onSearch: jest.fn(),
  buttonText: 'Search',
};

describe('Component SearchForm Snapshot', () => {
  it('Renders a SearchForm', () => {
    const tree = renderer.create(<SearchForm {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
