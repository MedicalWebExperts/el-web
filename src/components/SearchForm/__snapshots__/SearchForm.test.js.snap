// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SearchForm Snapshot Renders a SearchForm 1`] = `
<div
  className="jsx-3270843699"
  style={Object {}}
>
  <div
    className="jsx-2910760301"
  >
    <input
      className="jsx-2910760301"
      name="name"
      onChange={[Function]}
      placeholder="name"
      style={Object {}}
      type="text"
      value=""
    />
  </div>
  <div
    className="jsx-2327929533"
  >
    <select
      className="jsx-2327929533"
      onChange={[Function]}
      style={Object {}}
      value=""
    >
      <option
        className="jsx-2327929533"
        default={true}
        disabled={true}
        value=""
      >
        animals
      </option>
      <option
        className="jsx-2327929533"
        value="tiger"
      >
        Tiger
      </option>
      <option
        className="jsx-2327929533"
        value="lion"
      >
        Lion
      </option>
      <option
        className="jsx-2327929533"
        value="panther"
      >
        Panther
      </option>
      <option
        className="jsx-2327929533"
        value="jaguar"
      >
        Jaguar
      </option>
    </select>
  </div>
  <div
    className="jsx-2327929533"
  >
    <select
      className="jsx-2327929533"
      onChange={[Function]}
      style={Object {}}
      value=""
    >
      <option
        className="jsx-2327929533"
        default={true}
        disabled={true}
        value=""
      >
        gender
      </option>
      <option
        className="jsx-2327929533"
        value="male"
      >
        Male
      </option>
      <option
        className="jsx-2327929533"
        value="female"
      >
        Female
      </option>
    </select>
  </div>
  <div
    className="jsx-2888778090"
  >
    <button
      className="jsx-2888778090 outline"
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      Advanced Search
    </button>
  </div>
  <div
    className="jsx-2888778090"
  >
    <button
      className="jsx-2888778090 default"
      onClick={[Function]}
      style={Object {}}
      type="button"
    >
      Search
    </button>
  </div>
</div>
`;
